### ISSUE TITLE
Variable with empty list causes "No declaration found" on submodules

### ISSUE THREAD

**Issue by FeltlP** [hashicorp/vscode-terraform#1592]
### Extension Version

2.28.1

### VS Code Version

Version: 1.83.0 (Universal)
Commit: e7e037083ff4455cf320e344325dacb480062c3c
Date: 2023-10-03T16:13:15.449Z (1 wk ago)
Electron: 25.8.4
ElectronBuildId: 24154031
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 22.6.0

### Operating System

macOS Ventura 13.6 (22G120)

### Terraform Version

Terraform v1.6.1 on darwin_arm64

### Steps to Reproduce

Having the following structure: (included in the zip)

```
.
├── main.tf
└── modules
    ├── submodule-a
    │   ├── main.tf
    │   └── variables.tf
    └── submodule-b
        └── variables.tf
```

[no-declaration-found.zip](https://github.com/hashicorp/vscode-terraform/files/12883364/no-declaration-found.zip)

### Expected Behavior

No error, as on version 2.27.2, or to be consistent with terraform validate

### Actual Behavior

terraform validate shows no issues, VSCode with extension 2.28.1 shows "No declaration found" on modules/submodule-a/main.tf line 4

### Project Structure

```sh
.
├── main.tf
└── modules
    ├── submodule-a
    │   ├── main.tf
    │   └── variables.tf
    └── submodule-b
        └── variables.tf
```

### Workarounds

Define the type explicitly on all variable files as type = list(string)

### References

#1576 

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by CapricornVortex** [comment]
I also saw this yesterday - Same issue as above, I set the type as   type = list(any) but I still had the error - popped a value in for the default empty list and saved : 
default = ["value"]
and then removed it and resaved - the "No declaration found" error has now gone even though the code is now the same

**Comment by FeltlP** [comment]
@CapricornVortex, also seen this. For me it often has some delay between save & showing up the error again (or the other way round).
The most reliable way to validate if the error is gone, was to change the plugin version (2.27.2 <> 2.28.1) & reload

**Comment by mario-nc** [comment]
<img width="739" alt="image" src="https://github.com/hashicorp/vscode-terraform/assets/110963478/a7d14feb-7616-49d4-9ee0-09a17620f60e">

Pretty same problem for me. It does not complain when `var.tgw_id` is referenced under `transit_gateway_id`, however it it shows error (No declaration found for "var.tgw_id") when referenced in `depends_on` clause.

**Comment by radeksimko** [comment]
@FeltlP I was able to reproduce this but haven't had the chance to debug it yet. Thanks for the report and for attaching the ZIP file, that's helpful.

**Comment by mskrishna2808** [comment]
2.27.2 VS Code extension version is fixed issue for me

**Comment by dbanck** [comment]
We just released a new version [2.29.0](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.29.0). This fixes the reported bug. The update should appear automatically in VS Code.

Thank you for the report @FeltlP 

--- 

In case you experience any different validation related bug, please do let us know through a new issue.
